test_dnsutil.py

```diff
@@ -23,3 +23,45 @@ def _rc(content: str | bytes) -> tuple[list[str], list[str]]:
     assert _rc(b"nameserver\t1::0\xca4\nnameserver 1.2.3.4  ") == (["1.2.3.4"], [])
     assert _rc("search foo.com bar.com\nnameserver  1.2.3.4\n") == (["1.2.3.4"], ["foo.com", "bar.com"])
     assert _rc("search foo.com bar.com\n\nsearch xxx\nnameserver  1.2.3.4\n") == (["1.2.3.4"], ["xxx"])
+
+
+def test_dnsmasq_servers_parse() -> None:
+
+    content, entries = dnsutil._dnsmasq_servers_content_update(b"", "cluster1", "192.168.122.2")
+    assert content == (
+        b'# Written by cluster-deployment-automation for resolving cluster names.\n'
+        b'# This file is passed to dnsmasq via the --servers-file= option\n'
+        b'#\n'
+        b'# Reload after changes with\n'
+        b'#   systemctl restart dnsmasq.service\n'
+        b'#   pkill -HUP -f \'\\<dnsmasq\\>.*conf-file=/var/lib/libvirt/dnsmasq/default.conf\'\n'
+        b'server=/cluster1.redhat.com/192.168.122.2\n'
+    )
+
+    content2, entries2 = dnsutil._dnsmasq_servers_content_update(content, "cluster1", "192.168.122.2")
+    assert content == content2
+
+    assert dnsutil._dnsmasq_servers_content_parse(content) == [b"server=/cluster1.redhat.com/192.168.122.2"]
+
+    content2, entries2 = dnsutil._dnsmasq_servers_content_update(content, "cluster2", "192.168.123.2")
+    assert content2 == (
+        b'# Written by cluster-deployment-automation for resolving cluster names.\n'
+        b'# This file is passed to dnsmasq via the --servers-file= option\n'
+        b'#\n'
+        b'# Reload after changes with\n'
+        b'#   systemctl restart dnsmasq.service\n'
+        b'#   pkill -HUP -f \'\\<dnsmasq\\>.*conf-file=/var/lib/libvirt/dnsmasq/default.conf\'\n'
+        b'server=/cluster1.redhat.com/192.168.122.2\n'
+        b'server=/cluster2.redhat.com/192.168.123.2\n'
+    )
```

Comment:
Shouldn't the assert check if there is any line that contains `server=/cluster1.redhat.com/192.168.122.2` ?
The test now relies on no typos in the string and this means that if we change the string (which is not going to break the functionality), you will also break the test.

Reply:
> Shouldn't the assert check if there is any line

Maybe. I can adjust that.

> The test now relies on no typos in the comment and this means that if we change the comment (which is not going to break the functionality), you will also break the test.

I would rather say: if we change the comment, we will need to adjust the test too. Changing the comment does not happen by accident, it's a patch that does that. Then the test also needs adjustment. Which is easy to do, and impossible to miss.

Anyway. I will change that.